gobblin-data-management/src/test/java/org/apache/gobblin/data/management/dataset/ManifestBasedDatasetFinderTest.java

```diff
@@ -173,6 +174,34 @@ public void testIgnoreFilesWithSamePermissions() throws IOException, URISyntaxEx
     }
   }
 
+  @Test
+  public void testFindDatasetEmptyRoot() throws Exception {
+    //Get manifest Path
+    String manifestLocation = getClass().getClassLoader().getResource("manifestBasedDistcpTest/manifestRootDirEmpty.json").getPath();
+
+    // Test manifestDatasetFinder
+    Properties props = new Properties();
+    props.setProperty("gobblin.copy.manifestBased.manifest.location", manifestLocation);
+    props.setProperty(ConfigurationKeys.DATA_PUBLISHER_FINAL_DIR, "/");
+    ManifestBasedDatasetFinder finder = new ManifestBasedDatasetFinder(localFs, props);
+    List<ManifestBasedDataset> datasets = finder.findDatasets();
+    Assert.assertEquals(datasets.size(), 1);
+    try (
+        FileSystem sourceFs = Mockito.mock(FileSystem.class);
```

Comment:
just curious, do mocks require a close ?

Reply:
No because it's try with resources https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html

Reply:
i was saying that probably we do not need to close mocked classes, because there are no resources...